src/transformation/visitors/language-extensions/operators.ts

```diff
@@ -84,25 +81,15 @@ function getOperatorMapExtensionKindForCall(context: TransformationContext, node
     if (!typeDeclaration) {
         return;
     }
-    const mapping = extensions.getExtensionKind(typeDeclaration);
-    if (mapping !== undefined && operatorMapExtensions.has(mapping)) {
-        return mapping;
-    }
-}
-
-function isOperatorMapDeclaration(declaration: ts.Declaration) {
-    const typeDeclaration = getTypeDeclaration(declaration);
-    if (typeDeclaration) {
-        const extensionKind = extensions.getExtensionKind(typeDeclaration);
-        return extensionKind !== undefined ? operatorMapExtensions.has(extensionKind) : false;
-    }
+    const type = context.checker.getTypeFromTypeNode(typeDeclaration.type);
+    return operatorMapExtensions.find(extensionKind => extensions.isExtensionType(type, extensionKind));
 }
 
 function isOperatorMapType(context: TransformationContext, type: ts.Type): boolean {
     if (type.isUnionOrIntersection()) {
         return type.types.some(t => isOperatorMapType(context, t));
     } else {
-        return type.symbol?.declarations?.some(isOperatorMapDeclaration);
+        return operatorMapExtensions.some(extensionKind => extensions.isExtensionType(type, extensionKind));
```

Comment:
I don't understand why this change is required (in combination with the removal of the Set above). This turns this function from O(1) into O(n) (n being the nr of supported map extensions). Since this function is (I think?) executed for all function and method calls, that seems like a big performance hit I would like to avoid if possible.

Reply:
I agree the increased complexity is not ideal here. The issue comes down to the change in how extensions are identified. Previously it was by name which was easy to directly map to a kind and check for in a set. Now it's by brand which is a differently named property on each type so we have to iterate through the kinds we care about and test for each brand. Let me think about this a bit - there may be a way to refactor things to get the complexity back down.